demo/hostvolume/host.volume.hcl

```diff
@@ -0,0 +1,14 @@
+# Copyright (c) HashiCorp, Inc.
+# SPDX-License-Identifier: BUSL-1.1
+name      = "test"
+type      = "host"
+plugin_id = "example-host-volume"
+#plugin_id    = "mkdir"
+capacity_min = "50mb"
+capacity_max = "50mb"
+parameters {
+  a = "ayy"
+}
+
+# TODO(1.10): don't require node_pool
+node_pool = "default"
```

Comment:
I rebased on the target feature branch, and still get:

```
$ n volume create host.volume.hcl
Error creating volume: Unexpected response code: 500 (host volume insert: missing value for index 'node_pool')
```

looks to me like it is being checked against an _existing_ volume, or if node_pool _is_ set, then it makes sure that pool exists, but it's not defaulting to "default" anywhere.

especially problematic is that it fails to save state only after the plugin has run and created a volume on the node, which is now invisible to Nomad, so a user would need to survey their nodes to find such orphaned "disk"s.

Reply:
Ok, so that's bubbling up from the state store. That's weird... the RPC handler should be setting the node pool from the node it selected. I'll check on that in https://github.com/hashicorp/nomad/pull/24518

Reply:
Fixed in #24518... we weren't setting in the `createVolume` workflow but even if we did we want to make sure it's being set in the state store so that the Register RPC works with a NodeID but no NodePool in the request. I've adjusted the API tests to exercise that code path better too.